### PR TITLE
 Enable call counts in spacetime by default

### PR DESCRIPTION
**Pull request by lpw25**
#1180 added support for "call counts" in spacetime. This support is siabled by default is controlled by an `-spacetime-call-counts` configure-time option that both enables spacetime and enables call counts. 

This seems like the wrong default: there is little cost to having call counts and they are quite useful. It also seems slightly inconsistent with similar configure-time switches (e.g. `-disable-libunwind`).

This PR changes it so that call counts are enabled by default and are controlled by a `-disable-call-counts` configure-time switch.

Ideally this would go in 4.06, since the `-spacetime-call-counts` option has not yet appeared in a release.

### CODE REVIEW

File: utils/config.mli
Comment (damiendoligez) [inline utils/config.mli]: You're changing the interface of `config.ml` (and also of `configure`) for no good reason that I can see. Why not just do what the title of the PR says, and change the default value of the existing variable? (and of course add a `-no-spacetime-call-counts` option to `configure`)

**Review by damiendoligez — CHANGES_REQUESTED** [review]
I think the patch could be a lot smaller.

File: utils/config.mli
Comment (lpw25) [reply utils/config.mli]: Currently `with_spacetime_call_counts` means `with_spacetime && not disable_call_counts`. This isn't really consistent with other configure options like `disable_libunwind`, and it means that just changing the default of `with_spacetime_call_counts` to `true` would enable spacetime by default.

Comment (lpw25) [reply utils/config.mli]: In other words, I'm changing the name of this value in `config.ml` because I'm changing it's meaning, and I'm changing it's meaning to be consistent with both the other similar values in `config.ml` and with how it will appear in the arguments to `./configure.sh`.

Comment (damiendoligez) [reply utils/config.mli]: In `configure` yes, you're changing the meaning of the command-line option. In `config.ml` and in the C code, you have just added and removed negations, which (in my eye) is noise.

The variable itself never had the same meaning as the configure option, as can plainly be seen in `configure` itself.

Comment (damiendoligez) [reply utils/config.mli]: I'm also not a big fan of negative configuration variables, as these often lead to double-negated `#ifdef`s that make the code harder to read.

Comment (lpw25) [reply utils/config.mli]: > The variable itself never had the same meaning as the configure option, as can plainly be seen in configure itself.

I'm not sure I'm following this. The variable would previously be set precisely when `spacetime-call-counts` was passed to configure, which would also always imply `with_spacetime`. This seems to have the same meaning between variable and configure option. Or do you mean that the implication that `with_spacetime` was true was never used -- as evidenced by my only needing to negate various tests in this patch.

I'm happy to make the option positive -- but I would still rather rename it to something like `Config.enable_call_counts` in ml and `ENABLE_CALL_COUNTS` in c. In particular, I think that the `WITH_SPACETIME_CALL_COUNTS` is a bad name for the variable as it heavily implies that `WITH_SPACETIME` is true if it is true.

**Comment by lpw25** [comment]
I've rebased onto the new version of #1407 and changed the variables from `disable_call_counts` to `enable_call_counts`.

File: utils/config.mli
Comment (damiendoligez) [reply utils/config.mli]: What I mean is that the `configure` code says that `-spacetime-call-counts` means (with-spacetime AND with-spacetime-call-counts) which is obviously not the same as just with-spacetime-call-counts.

But anyway I'm OK with `enable_call_counts`. I don't much like `with_spacetime_call_counts` either.

File: configure
Comment (damiendoligez) [inline configure]: Surely the option should be `-enable-call-counts`.

**Review by damiendoligez — APPROVED** [review]
I'm OK with this version (except for the option name) but it's based on #1407 so you should wait for that PR to land before you merge this one.

File: configure
Comment (lpw25) [reply configure]: This line should be `enable_call_counts=false`. I still want the default to be that they are enabled, so the option here is still called `disable-call-counts`.

Comment (damiendoligez) [reply configure]: Sure. I was blindly assuming that the code was right.

**Comment by lpw25** [comment]
Since they no longer overlap, I've rebased this one off of #1407 so it can now be merged separately.

**Comment by damiendoligez** [comment]
Appveyor fails because you need to change `WITH_SPACETIME_CALL_COUNTS` to `ENABLE_CALL_COUNTS` in all four of the `config/Makefile.m*` files.

**Comment by lpw25** [comment]
Fixed and now the CIs pass (except the changes test which just seems to ignore the `no-change-entry-needed` label even when I restart the test explicitly)

**Comment by lpw25** [comment]
Merged and cherry-picked to 4.06 (175f55c)

**Pull request merged**
